### PR TITLE
Extraction of location data from the android device

### LINKED ISSUE #186: Extraction of location data from the android device
**Is your feature request related to a problem? Please describe.**
Extraction of location data.
Implementation from the command line to extract location data directly.
Creating API which will give the Location data(this will help in frontend).

### PR DESCRIPTION
**Pull request by swapnalshahil**
# Description

Extraction of location data.
Implementation from the command line to extract location data directly.
Creating API which will give the Location data(this will help in frontend).
API : /data/<case_name>/location
this will provide path to open 1) savedlocation.tsv 2)searchedlocation.tsv

Fixes #186 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

POSTMAN and CMD

CMD
2021-06-22 (4)_LI
2021-06-22 (5)_LI

savedlocation.tsv(above)

searchedlocation.tsv(above)

Extraction using POSTMAN (above)

API (above)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by swapnalshahil** [comment]
“gmm_storage.db" database stores search terms and possible navigation history amongst other
proprietary content in BLOB data records.  So even after converting to tsv some remains encoded. But we can see there searched terms, and links there which may be sufficient for us.

"gmm_myplaces.db" stores saved places,labelled places,etc. We can get links to those locations with other data also in savedlocation.tsv.

**Comment by swapnalshahil** [comment]
@shivanshu1333 sir! Please review! 🙂

**Comment by shivanshuraj1333** [comment]
@swapnalshahil either we've to find a way to extract proprietary content if it is possible or we have to keep only the useful information in tsv file which is readable by humans, in this case we need to parse the tsv file to extract only readable strings.

**Comment by swapnalshahil** [comment]
@shivanshu1333  sir! I implemented logic to get only letters in the searched location which is better than before.

**Comment by swapnalshahil** [comment]
Now it stores all bundles.

File: apiUtility/apiUtils.py [hunk 1]
```diff
@@ -17,8 +17,10 @@
 from scripts.message import store_sms_data, store_sms_messages
```
Comment (shivanshuraj1333) [inline apiUtility/apiUtils.py:17]: Add relevant comments at the top and above each function as discussed earlier

**Review by shivanshuraj1333 — CHANGES_REQUESTED** [review]

File: apiUtility/apiUtils.py (hunk 1 above)
Comment (swapnalshahil) [reply apiUtility/apiUtils.py:17]: Comments added to location.py

Comment (swapnalshahil) [reply apiUtility/apiUtils.py:17]: I will comment on other files separately sir!

**Comment by shivanshuraj1333** [comment]
@swapnalshahil we'll setup a meeting to discuss this

**Comment by swapnalshahil** [comment]
> @swapnalshahil we'll setup a meeting to discuss this

ok, sir!

**Review by shivanshuraj1333 — APPROVED** [review]
merging to master

**Pull request merged**